src/Bridge/Laravel/DbToolsServiceProvider.php

```diff
@@ -0,0 +1,270 @@
+                    $loader = match ($ext) {
+                        'php' => new PhpFileLoader($file, $connectionName),
+                        'yml', 'yaml' => new YamlLoader($file, $connectionName),
+                        default => throw new ConfigurationException(\sprintf(
+                            "File extension \"%s\" is unsupported (given path: \"%s\").", $ext, $file
+                        )),
+                    };
+
+                    $factory->addConfigurationLoader($loader);
+                }
+
+                foreach ($config->get('db-tools.anonymization', []) as $connectionName => $array) {
+                    $factory->addConfigurationLoader(new ArrayLoader($array, $connectionName));
+                }
+            }
+        );
+
+        $this->app
+            ->when(AnonymizerRegistry::class)
+            ->needs('$paths')
+            ->give(function (Application $app) {
+                // Validate user-given anonymizer paths.
+                $anonymizerPaths = $app->make('config')->get('db-tools.anonymizer_paths');
+                foreach ($anonymizerPaths as $path) {
+                    if (!\is_dir($path)) {
+                        throw new \InvalidArgumentException(\sprintf(
+                            '"db_tools.anonymizer_paths": path "%s" does not exist', $path
+                        ));
+                    }
+                }
+                // Set the default anonymizer directory only if the folder
+                // exists in order to avoid "directory does not exist" errors.
+                $defaultDirectory = $app->basePath('app/Anonymizer');
+                if (\is_dir($defaultDirectory)) {
+                    $anonymizerPaths[] = $defaultDirectory;
+                }
+            })
+        ;
+
+        /*$this->app->singleton(BackupperFactory::class, function (Application $app) {
+            return new BackupperFactory(
+                $app->make(DatabaseSessionRegistry::class),
+                $app->make(ConfigurationRegistry::class),
+                // Logger
+            );
+        });*/
+
+        /*$this->app->singleton(RestorerFactory::class, function (Application $app) {
+            return new RestorerFactory(
+                $app->make(DatabaseSessionRegistry::class),
+                $app->make(ConfigurationRegistry::class),
+                // Logger
+            );
+        });*/
+
+        /*$this->app->singleton(StatsProviderFactory::class, function (Application $app) {
+            return new StatsProviderFactory($app->make(DatabaseSessionRegistry::class));
+        });*/
+
+        /*$this->app
+            ->when([
+                AnonymizatorFactory::class,
+                BackupperFactory::class,
+                RestorerFactory::class,
+                StatsProviderFactory::class
+            ])
+            ->needs(LoggerInterface::class)
+            ->give(fn (Application $app) => $app->make('log'))
+        ;*/
```

Comment:
why is this commented ?

Reply:
I wrote it before realizing it was probably not necessary. I did not remove it for now. I would like to be sure there is no interest in keeping this code.

Reply:
Not sure to understand how it could work without it but I trust you ^^

Reply:
@SimonMellerin https://laravel.com/docs/11.x/container#zero-configuration-resolution

Reply:
@SimonMellerin As I let you know through the link above, Laravel will be able to instantiate classes in question in the commented code, since we tell it how to resolve their dependencies.

But if I did not misunderstand, Laravel will create a new instance each time an instance of those classes will be necessary.

I'm wondering if it is better that I restore the commented bindings to ensure that each instance will be a singleton.

What do you think?

Reply:
Well I would say that those services was designed to be singleton (as it was first designed to be used in a symfony container). Thus, to avoid any side effect I think you are right. Go for singleton !